examples/audio_embedder/android/app/src/main/java/com/google/mediapipe/examples/audioembedder/MainActivity.kt

```diff
@@ -0,0 +1,225 @@
+                    btnCompare.setOnClickListener {
+                        compareTwoAudioFiles()
+                    }
+                    btnPickAudioOne.setOnClickListener {
+                        selectAudioPos = 1
+                        getContent.launch("audio/*")
+                    }
+                    btnPickAudioTwo.setOnClickListener {
+                        selectAudioPos = 2
+                        getContent.launch("audio/*")
+                    }
+                    btnPlayAudioOne.setOnClickListener {
+                        uriOne?.let { it1 -> playAudio(it1) }
+                    }
+                    btnPlayAudioTwo.setOnClickListener {
+                        uriTwo?.let { it1 -> playAudio(it1) }
+                    }
+                }
+            }
+        }
+    }
+
+    private fun initBottomSheetControls() {
+        binding.bottomSheetLayout.spinnerDelegate.setSelection(
+            audioEmbedderHelper.currentDelegate, false
+        )
+        binding.bottomSheetLayout.spinnerDelegate.onItemSelectedListener =
+            object : AdapterView.OnItemSelectedListener {
+                override fun onItemSelected(
+                    parent: AdapterView<*>?,
+                    view: View?,
+                    position: Int,
+                    id: Long
+                ) {
+                    audioEmbedderHelper.currentDelegate = position
+                    resetEmbedder()
+                }
+
+                override fun onNothingSelected(parent: AdapterView<*>?) {
+                    /* no op */
+                }
+            }
+    }
+
+    // Compare two audio files here
+    private fun compareTwoAudioFiles() {
+        uriOne?.let { audio1 ->
+            uriTwo?.let { audio2 ->
+                // show progress bar
+                binding.flProgress.visibility = View.VISIBLE
+                backgroundExecutor.execute {
+                    audioEmbedderHelper.compare(
+                        audio1.createAudioData(this@MainActivity),
+                        audio2.createAudioData(this@MainActivity)
+                    )?.let { resultBundle ->
+                        updateResult(resultBundle)
+                    } ?: runOnUiThread {
+                        binding.flProgress.visibility = View.GONE
+                    }
+                }
+            }
+        }
+    }
+
+    private fun updateResult(resultBundle: AudioEmbedderHelper.ResultBundle) {
+        runOnUiThread {
+            binding.flProgress.visibility = View.GONE
+            binding.tvTitle.visibility = View.INVISIBLE
+            binding.tvSimilarity.visibility = View.VISIBLE
+            binding.tvSimilarity.text = String.format(
+                "Similarity: %.2f", resultBundle.similarity
+            )
+            binding.bottomSheetLayout.inferenceTimeVal.text =
+                String.format("%d ms", resultBundle.inferenceTime)
+        }
+    }
+
+    // Reset Embedder.
```

Comment:
This comment is redundant given the function name. We should add more comments throughout the class to highlight what's happening, however.